### DISCUSSION TITLE
v1.80.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#4303] [Announcements]
# V1.80.0

We urge all Immich users to update this release as soon as possible, as it fixes a severe security vulnerability caused by a third-party dependency. See below for more details.

> **Warning**
> ### **Breaking Changes** (Action Required)
> - The reverse geocoding settings have been moved to the admin settings pages. If you have set the level of accuracy in `.env` file, you will need to make the corresponding changes in the settings menu on the administration page on the web.
> - Migrate thumbnails to a new folder structure by running the new migration job `Admin > Settings > Migration`. See #4112 for more details)

## Highlights

Welcome to release `v1.80.0` of Immich. I just returned from a long vacation to visit my family in Vietnam. The team has been working very productively while I was not actively involved in the development. This makes me happy and proves the project is in good hands. 

I am excited to share this release with you, emphasizing bug fixing of the recent new `library` feature and many further QoL improvements. Below are some of the highlights

- Added mechanism to store generated files for `thumbnails` and `encoded-videos` in subfolders.
- Added an ability to suggest people’s names during name assignment for faces.
- Added an option to show the sidebar menu for the people page in the user setting on the web.
- Added week number as an option for path templating.
- Bugfixes related to External Libraries, metadata processing, and Live Photos

### Storing generated files for `thumbnails` and `encoded-videos` in subfolders

This feature helps improve the performance by storing generated files in subdirectories. This yields smaller directory sizes (in terms of files per directory), decreasing the size of the inode entry.

New asset and face thumbnails will be stored at i.e. `89/ab/`, if their id starts with `89abxxxx-xxxx...`

We added a `Migration` job on the Jobs page to migrate the existing directory to this new structure.

<img width="917" alt="Screenshot 2023-09-17 at 21 38 26" src="https://github.com/immich-app/immich/assets/36593685/a71777ed-1410-435e-9d62-b3e75d874330">

The job will perform the following actions:

- Migration job to migrate all files to new folder structure (and possibly more migration stuff in the future)
- Clean up directories if assets/people get deleted

### Suggesting people’s names during name assignments for faces

https://github.com/immich-app/immich/assets/74269598/2588b940-cc84-4347-a52a-3ff00e98f6fa

### Option to show the sidebar menu for the people page in the user setting on the web.
Add a new account preference`Account Settings > Sidebar Settings`, which allows a user to enable a sidebar link to "People"

| Light | Dark
| - | - |
| ![image](https://github.com/immich-app/immich/assets/4334196/0d1ae058-6af2-461b-a449-ee8d41545f50) | ![image](https://github.com/immich-app/immich/assets/4334196/13e0a0d6-f31e-464d-af2d-edbe872708cc) | 

![image](https://github.com/immich-app/immich/assets/4334196/86470458-aa18-4fb2-b920-a7868f08d3f2)
![image](https://github.com/immich-app/immich/assets/4334196/4f5a0e80-a907-403a-a9df-0aa73f160d7c)

### Option to use week number as an option for path templating

You can now use week numbers ('W') and padded week numbers ('WW') patterns for storage templating.

<img width="823" alt="Screenshot 2023-09-28 at 18 59 17" src="https://github.com/immich-app/immich/assets/36593685/393896e2-455b-44ac-9f18-89a51443b801">

<img width="838" alt="Screenshot 2023-09-28 at 18 58 20" src="https://github.com/immich-app/immich/assets/36593685/34fca408-77ad-40c6-a33b-cd07c5925c64">

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

### Security
* Fixes a security vulnerability ([CVE-2023-5129](https://securityaffairs.com/151576/hacking/cve-2023-5129-libwebp-flaw.html)) by rebuilding the Docker image with the newest Debian libwebp library

### Web
* feat(web): suggest people when typing a name by @martabal in https://github.com/immich-app/immich/pull/4126
* feat(web): people sidebar link by @jrasm91 in https://github.com/immich-app/immich/pull/4257
* feat(web): better context menu position by @jrasm91 in https://github.com/immich-app/immich/pull/4271
* feat(web): version links to releases page by @danieldietzler in https://github.com/immich-app/immich/pull/4288
* fix(web): escape shortcut by @martabal in https://github.com/immich-app/immich/pull/3753
* fix(web): Context menu underflowing on people page by @danieldietzler in https://github.com/immich-app/immich/pull/4270
* refactor(web): album listing page by @martabal in https://github.com/immich-app/immich/pull/4146
* chore(web): remove flowbite by @alextran1502 in https://github.com/immich-app/immich/pull/4178
* chore(web): remove upload preview by @jrasm91 in https://github.com/immich-app/immich/pull/4219
* chore(web): use axios isCancel function by @uhthomas in https://github.com/immich-app/immich/pull/4227

### Server
* feat(server, web)!: Move reverse geocoding settings to the UI by @danieldietzler and @jrasm91 in https://github.com/immich-app/immich/pull/4222
* feat(server): random assets API by @daniele-athome in https://github.com/immich-app/immich/pull/4184
* feat(server): Add support for Photoshop files by @mros123 in https://github.com/immich-app/immich/pull/4192
* feat(server): Add week numbers for templating by @danieldietzler in https://github.com/immich-app/immich/pull/4263
* fix(server): use mtime instead of ctime for fileCreatedAt by @daniele-athome in https://github.com/immich-app/immich/pull/4191
* fix(server): Random assets endpoint return always n assets if available by @danieldietzler in https://github.com/immich-app/immich/pull/4201
* fix(server): require library.write to upload assets to library by @danieldietzler in https://github.com/immich-app/immich/pull/4200
* fix(server): search for terms separated by hyphens and/or underscores in asset search by @ochen1 in https://github.com/immich-app/immich/pull/4156
* fix(server): metadata extraction error by @alextran1502 in https://github.com/immich-app/immich/pull/4210
* fix(server): android live motion by @jrasm91 in https://github.com/immich-app/immich/pull/4218
* fix(server): use srgb pipeline for srgb images by @mertalev in https://github.com/immich-app/immich/pull/4101
* fix(server): tests by @jrasm91 in https://github.com/immich-app/immich/pull/4229
* fix(server): handle NaN in metadata extraction by @djak250 in https://github.com/immich-app/immich/pull/4221
* fix(server): Thumbnail migration creating unnecessary directories by @danieldietzler in https://github.com/immich-app/immich/pull/4251
* fix(server): android motion photo by @jrasm91 in https://github.com/immich-app/immich/pull/4254
* fix(server): live photo linking by @jrasm91 in https://github.com/immich-app/immich/pull/4253
* fix(server): Exclude archived assets from search-explore #4041 by @sellnat77 in https://github.com/immich-app/immich/pull/4122
* fix(server): always disable two-pass mode for video thumbnails by @mertalev in https://github.com/immich-app/immich/pull/4258
* fix(server): handle number lists in metadata extraction by @jrasm91 in https://github.com/immich-app/immich/pull/4273
* fix(server): cli import by @etnoy in https://github.com/immich-app/immich/pull/4224
* refactor(server): person thumbnail job by @jrasm91 in https://github.com/immich-app/immich/pull/4233
* refactor(server): move metadata extraction to domain by @danieldietzler in https://github.com/immich-app/immich/pull/4243
* refactor(server): merge facial-recognition and person by @danieldietzler in https://github.com/immich-app/immich/pull/4237
* chore(server): unit tests for metadata service by @danieldietzler in https://github.com/immich-app/immich/pull/4280
* chore(server): Store generated files (thumbnails, encoded video) in subdirectories by @danieldietzler in https://github.com/immich-app/immich/pull/4112

### Mobile
* feat(mobile): use map settings from server-config by @shalong-tanwen in https://github.com/immich-app/immich/pull/4045
* fix(mobile): reduce server version api calls by @shalong-tanwen in https://github.com/immich-app/immich/pull/4265
* fix(mobile): assetList is empty by @fujie-xiyou in https://github.com/immich-app/immich/pull/4213
* fix(mobile): speed up RenderList creation for timeline by @fyfrey in https://github.com/immich-app/immich/pull/4103

### Others
* chore: Correct admin commands by @erichelgeson in https://github.com/immich-app/immich/pull/4177
* docs: deprecate read only assets by @etnoy in https://github.com/immich-app/immich/pull/4226
* docs: use non-conflicting port to serve docs by @jrasm91 in https://github.com/immich-app/immich/pull/4230
* docs: add Japanese translated README by @eltociear in https://github.com/immich-app/immich/pull/4268

## New Contributors
* @erichelgeson made their first contribution in https://github.com/immich-app/immich/pull/4177
* @mros123 made their first contribution in https://github.com/immich-app/immich/pull/4192
* @ochen1 made their first contribution in https://github.com/immich-app/immich/pull/4156
* @fujie-xiyou made their first contribution in https://github.com/immich-app/immich/pull/4213
* @djak250 made their first contribution in https://github.com/immich-app/immich/pull/4221

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.79.1...v1.80.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.80.0'>v1.80.0</a>.</em>

**Comment by nep2ner** [comment]
Thanks for getting libwebp updated so quickly

**Comment by lordraiden** [comment]
I don't understand  do we need to have the same setting in both places or just remove the env variable and use the admin setting in the webui?

Thanks

```

###################################################################################
# Reverse Geocoding
#
# Reverse geocoding is done locally which has a small impact on memory usage
# This memory usage can be altered by changing the REVERSE_GEOCODING_PRECISION variable
# This ranges from 0-3 with 3 being the most precise
# 3 - Cities > 500 population: ~200MB RAM
# 2 - Cities > 1000 population: ~150MB RAM
# 1 - Cities > 5000 population: ~80MB RAM
# 0 - Cities > 15000 population: ~40MB RAM
####################################################################################

DISABLE_REVERSE_GEOCODING=false
REVERSE_GEOCODING_PRECISION=3
```

**Reply by jrasm91** [reply]
These aren't used anymore, you only need to adjust the web UI now

**Comment by geekykid12** [comment]
I got a notification of the update and installed it and am having some issues now and looking for advice.
Ran the following
`sudo docker compose down &&
sudo docker compose pull &&
sudo docker compose up -d`
After my machine Learning Container keeps restarting
`docker logs immich_machine_learning` reveals the following error.
![Screenshot_20231002-181408](https://github.com/immich-app/immich/assets/11451677/3736df59-94fe-49a9-8d85-5025758e2f34)

**Reply by jrasm91** [reply]
I think docker compose up -d doesn't seem to play very nice with machine learning. Try to stop it completely and then recreate it. If it still won't start, go ahead and delete the volume for the container and then run compose up again.

**Reply by geekykid12** [reply]
Is the appropriate method of stopping just docker compose down?

**Reply by geekykid12** [reply]
Encountered the same issue without -d. Also tried downgrading to v1.79.1 and updated python to 3.11.

**Reply by jrasm91** [reply]
Usually up -d works fine, but a few people are having issues with the machine learning container. Idk if it is related to running up while the stack is already running or not. Compose pull, down, and up is probably the safest though.

**Reply by geekykid12** [reply]
Still having issues. Ran docker compose pull, docker compose down then docker compose up and it's the same error message on the logs.

**Reply by geekykid12** [reply]
Got it to start by doing docker compose up -d, deleting the container in /var/lib/docker/containers and then it started on its own.

**Reply by geekykid12** [reply]
I'd like to upgrade but will wait until the server is under less load

**Reply by djak250** [reply]
You could try with `docker compose up --force-recreate` I've similar issues with other containers where it should've done the recreate by default, but forcing it fixed it.

**Comment by wutr** [comment]
> * Migrate thumbnails to a new folder structure by running the new migration job `Admin > Settings > Migration`. See

Should be `Administration > Jobs > Migration`

Worked great once I found it though!

Edit: unless this was previously correct and subsequently moved somewhere between 1.80.0 and 1.82.0

**Comment by Huber1** [comment]
A question concerning the geocoding precision. Is there still a way to set it?
I used to not have it set to highest precision, because I'm living in a larger city, but my location is nearer to the center of some small village nearby, so the location is wrong.
Also when searching, I don't know all small villages but just search for the bigger ones nearby
